### PR TITLE
Simplicity 4 regression tests

### PR DESCRIPTION
**Pull request by JanAdamiak**
This adds regression tests for all the pathways.
Each test is followed by a teardown function to clean-up the working directory.
Also added a GitHub Action that will run the test suite on both Ubuntu and Windows on each pull request.

### CODE REVIEW

**Comment by JanAdamiak** [comment]
@pydanny Let me know if there is anything missing in this PR!

**Comment by pydanny** [comment]
Your PR looks great! Running it through the paces now!

File: .github/workflows/test.yml [hunk 1]
```diff
@@ -0,0 +1,24 @@
+on:
+    pull_request:
+  
+jobs:
+    version_matrix:
+      strategy:
+        matrix:
+          os: [ubuntu-latest, windows-latest]
+          python-version: ['3.7', '3.8', '3.9', '3.10', '3.11']
```
Comment (pydanny) [inline .github/workflows/test.yml:9]: Python 3.7 is past end-of-life and should be removed.

**Review by pydanny — CHANGES_REQUESTED** [review]
Great PR! Just one problem, the cookiecutter dependency is missing in the `pyproject.toml` file.

**Comment by pydanny** [comment]
Great job! Just reviewed it and left some comments.

File: .github/workflows/test.yml (hunk 1 above)
Comment (JanAdamiak) [reply .github/workflows/test.yml:9]: Good catch!

**Comment by JanAdamiak** [comment]
Thanks! I've fixed all of the comments.

**Review by pydanny — APPROVED** [review]
LGTM, some weird windows errors though. 🤔 

I'll just merge it in and we can fix the windows issues in a future PR

**Pull request merged**